src/utils/urlHelpers.ts

```diff
@@ -107,11 +109,35 @@ export function getUrlFromState(
   state: RequestConfigs,
   options: QueryParamEncodingOptions = {},
 ): string {
-  const { baseUrl: url, origin, pathname } = options;
-  const baseUrl = url || `${origin}${pathname}`;
+  const { baseUrl: url, origin: initialOrigin, pathname: initialPathname } = options;
+  let origin = initialOrigin;
+  let pathname = initialPathname;
 
+  // If none of the options are provided
+  if (!url && !origin && !pathname) {
+    throw new Error('Either baseUrl or both origin and pathname must be provided in options');
+  }
+
+  // Extract origin and pathname from url if they are not provided
+  if (url && (!origin || !pathname)) {
+    const parsedUrl = new URL(url);
+    origin = parsedUrl.origin || '';
+    pathname = parsedUrl.pathname || '';
+  }
```

Comment:
Let's refactor this to only take in `url` in `options`. I don't see any reason why we should take in `origin` and `pathname` as parameters here.